services/api/src/resources/organization/resolvers.ts

```diff
@@ -765,4 +766,69 @@ export const addGroupToOrganization: ResolverFn = async (
   });
 
   return "success"
+};
+
+// removes a user from all groups in an organisation
+export const removeUserFromOrganizationGroups: ResolverFn = async (
+  _root,
+  { input: { user: userInput, organization: organizationInput } },
+  { models, sqlClientPool, hasPermission, userActivityLogger }
+) => {
+
+  if (R.isEmpty(userInput)) {
+    throw new Error('You must provide a user id or email');
+  }
+
+  const user = await models.UserModel.loadUserByIdOrUsername({
+    id: R.prop('id', userInput),
+    username: R.prop('email', userInput)
+  });
+
+  // check the organization exists
+  const organizationData = await Helpers(sqlClientPool).getOrganizationById(organizationInput);
+  if (organizationData === undefined) {
+    throw new Error(`Organization does not exist`)
+  }
+
+  // check permissions and get groups
+  await hasPermission('organization', 'viewGroup', {
+    organization: organizationInput,
+  });
+  const orgGroups = await models.GroupModel.loadGroupsByOrganizationId(organizationInput);
+
+  // iterate through groups and remove the user
+  let groupsRemoved = []
+  for (const group in orgGroups) {
+    if (R.prop('lagoon-organization',  orgGroups[group].attributes)) {
+      // if this is a group in an organization, check that the user removing members from the group in this org is in the org
+      await hasPermission('organization', 'addGroup', {
+        organization: R.prop('lagoon-organization', orgGroups[group].attributes)
+      });
+    } else {
+      await hasPermission('group', 'removeUser', {
+        group: orgGroups[group].id
+      });
+    }
+
+    try {
+      await models.GroupModel.removeUserFromGroup(user, orgGroups[group]);
+      groupsRemoved.push(orgGroups[group].name);
+    } catch (error) {
+      throw new Error(`Unable to remove user from group: ${error}`)
+    }
+  }
```

Comment:
I'd probably change this up a bit and do one permission check, then work out the groups the user is in that are in the organization, then remove them from those groups only.

```
  await hasPermission('organization', 'removeGroup', {
    organization: organizationInput,
  });
  const orgGroups = await models.GroupModel.loadGroupsByOrganizationId(organizationInput);

  // iterate through groups and work out which of this users groups are in the organization
  let groupsRemoved = []
  for (const group in orgGroups) {
    // if the groups organization is the one to remove from, push it to a new array
    if (R.prop('lagoon-organization',  orgGroups[group].attributes) == organizationInput) {
      groupsRemoved.push(orgGroups[group].name);
    }
  }
  // iterage over the groups to remove now (or use a model helper that takes the `groupsRemoved` as an input instead of doing the for loop here one by one, see note below)
  for (const group in groupsRemoved) {
    try {
      await models.GroupModel.removeUserFromGroup(user, groupsRemoved[group]);
    } catch (error) {
      throw new Error(`Unable to remove user from group: ${error}`)
    }
  }
```

What could be even better though is using a group model helper similar to this https://github.com/uselagoon/lagoon/blob/delete-organization/services/api/src/models/group.ts#L744-L785 (which is from https://github.com/uselagoon/lagoon/pull/3507) that would remove the user from all provided groups, and then update the redis keycloak cache once complete

Reply:
https://github.com/uselagoon/lagoon/pull/3509/commits/b772bc0a71d598bc0eda8937b8b68e909ca92f21